internal/expr/query.go

```diff
@@ -100,6 +102,23 @@ func (pe *PreparedExpr) Query(args ...any) (ce *QueryExpr, err error) {
 	return &QueryExpr{outputs: pe.outputs, sql: pe.sql, args: qargs}, nil
 }
 
+type NullT struct {
+	Value reflect.Value
+	Valid bool
+}
+
+func (nt *NullT) Scan(src any) error {
+	if src == nil {
+		nt.Valid = false
+		return nil
+	}
+	err := convert.ConvertAssign(nt.Value.Addr().Interface(), src)
+	if err == nil {
+		nt.Valid = true
+	}
+	return err
```

Comment:
you never set `nt.Valid` to `false` in case of `err`. I would probably write it like this

```
nt.Valid = false
if src == nil {
    return nil
}
err := convert.ConvertAssign(nt.Value.Addr().Interface(), src)
if err == nil {
    nt.Valid = true
}
return err
```

Reply:
yeah, but then setting it to `false` when `src == nil` was inconsistent. Anyway, just nitpicking :-)